pxtlib/util.ts

```diff
@@ -1948,6 +1948,62 @@ namespace ts.pxtc.Util {
 
         return enabled;
     }
+
+    /**
+     * Remove potentially sensitive info from the given data to avoid logging it.
+     * Currently only supports string data.
+     */
+    export function cleanData(data: string): string;
+    export function cleanData(data: pxt.Map<string | number>): pxt.Map<string | number>;
+    export function cleanData(data: any): any {
+        if (!data) return data;
+
+        let result: any;
+        if (typeof data === "string") {
+            result = removePropertiesWithPossibleUserInfo(data);
+        } else if (typeof data === "object") {
+            result = {};
+            for (const [key, value] of Object.entries(data)) {
+                result[key] = typeof value === "string" ? removePropertiesWithPossibleUserInfo(value) : value;
+            }
+        } else {
+            result = data;
+        }
+
+        return result;
+    }
+
+    /**
+     * Attempts to remove commonly leaked PII
+     * @param property The property which will be removed if it contains user data
+     * @returns The new value for the property
+     * 
+     * Taken from https://github.com/microsoft/vscode/blob/main/src/vs/platform/telemetry/common/telemetryUtils.ts
+     */
+    function removePropertiesWithPossibleUserInfo(property: string): string {
+        // If for some reason it is undefined we skip it (this shouldn't be possible);
+        if (!property) {
+            return property;
+        }
+
+        const userDataRegexes = [
+            { label: 'Google API Key', regex: /AIza[A-Za-z0-9_\\\-]{35}/ },
+            { label: 'Slack Token', regex: /xox[pbar]\-[A-Za-z0-9]/ },
+            { label: 'GitHub Token', regex: /(gh[psuro]_[a-zA-Z0-9]{36}|github_pat_[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59})/ },
+            { label: 'Generic Secret', regex: /(key|token|sig|secret|signature|password|passwd|pwd|android:value)[^a-zA-Z0-9]/i },
+            { label: 'CLI Credentials', regex: /((login|psexec|(certutil|psexec)\.exe).{1,50}(\s-u(ser(name)?)?\s+.{3,100})?\s-(admin|user|vm|root)?p(ass(word)?)?\s+["']?[^$\-\/\s]|(^|[\s\r\n\\])net(\.exe)?.{1,5}(user\s+|share\s+\/user:| user -? secrets ? set) \s + [^ $\s \/])/ },
+            { label: 'Email', regex: /@[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+/ } // Regex which matches @*.site
```

Comment:
This email regex seems overly broad. It will match strings like "@foo.anything". Consider using `/\b[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,63}\b/i`. It's narrower, will early out sooner, and has a max scanning length.

It also doesn't match the name part of the email at all. That's works ok here since you're nuking the whole value, but not a great if we ever decide to keep some of the surrounding text.